### ISSUE TITLE
ansible_local: Destroy fails if guest lacks playbook

### ISSUE THREAD

**Issue by siddharthist** [mitchellh/vagrant#6763]
When destroying a machine built with the new `ansible_local` provisioner that hasn't sucessfully recieved its playbook, Vagrant will show the following message:
```
$ vagrant destroy -f
There are errors in the configuration of this machine. Please fix
the following errors and try again:

ansible local provisioner:
* `playbook` does not exist on the guest: /vagrant/terraform.sample.yml
```
The expected behavior is that Vagrant will destroy the VM, with or without the playbook existing. 

The full gist is [here](https://gist.github.com/siddharthist/d810f9c53e1af2c605d9).

**Comment by gildegoma** [comment]
@siddharthist Thanks for reporting this. Without digging into the code, I don't know why the `config` phase is executed on `vagrant destroy`. This is not something specific to the Ansible provisioners though, and I'd like to check how other provisioners could be affected (or how they handle this case).

No time now, but I hope we can fix it for 1.8.2 :)

**Comment by omidraha** [comment]
I have the same problem.
```
% vagrant destroy 
No vagrant-config.yml found, using defaults
    control-01: Are you sure you want to destroy the 'control-01' VM? [y/N] y
There are errors in the configuration of this machine. Please fix
the following errors and try again:
ansible local provisioner:
* `playbook` does not exist on the guest: /vagrant/sample.yml
```
Vagrent version:
```
% vagrant version   
Installed Version: 1.8.1
```
Before  Vagrant 1.8.2 release, how can i fix that manually?

**Comment by demoneaux** [comment]
After some tinkering, I did a `vagrant halt` before `vagrant destroy`, which allowed the box to be destroyed.

**Comment by gildegoma** [comment]
For the records: This issue is finally fixed by a7ee56459b2c990550c4fc33883c55cf4145f463, which is way better than the original fix (4e451c6)
